unittests.py

```diff
@@ -505,7 +505,7 @@ def _testFailure(self, cmdLine, expectedExceptionClass):
 
     def _testFull(self, cmdLine, expectedSourceFiles, expectedOutputFile):
         sourceFiles, outputFile = CommandLineAnalyzer.analyze(cmdLine)
-        self.assertEqual(sourceFiles, expectedSourceFiles)
+        self.assertEqual([s for s, _ in sourceFiles], expectedSourceFiles)
```

Comment:
Wouldn't it be nicer to rather adjust the caller of `_testFull` such that `expectedSourceFiles` is actually a list of tuples, specifying the expected language (if any) specification for every source file?